backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/docvalues/impl/DocValuesJoin.java

```diff
@@ -18,17 +21,118 @@
 import org.apache.lucene.util.BytesRef;
 
 /**
- * Allows to sort on nested document field values.
+ * Utils around docvalues and nested documents.
  * <p>
  * Copied and adapted from {@code org.elasticsearch.search.MultiValueMode} class
  * of <a href="https://github.com/elastic/elasticsearch">Elasticsearch project</a>.
  */
-public class OnTheFlyNestedSorter {
+public class DocValuesJoin {
 
-	private OnTheFlyNestedSorter() {
+	private DocValuesJoin() {
 	}
 
-	public static SortedDocValues sort(SortedDocValues selectedValues, BitSet parentDocs, DocIdSetIterator childDocs) {
+	public static SortedDocValues getJoinedAsSingleValuedSorted(LeafReaderContext context, String field,
```

Comment:
Coupling didn't change since the comparator source was already using this class.

High cohesion is a valid argument, but frankly if I have a util for longs and integers, I'd rather have it for all other types for consistency.
The end goal is to keep the code maintainable, and I doubt spreading similar code (which is basically copy/pasting with a few types changed) across multiple classes will make the code more maintainable.

If anything, the "clean" solution would be to introduce some abstraction over the docvalues. Which would add more commits to this PR, with more unrelated changes, which I think we can agree would not be great.

I also expect more changes to this class in the future, probably moving it to an instantiable class similar to `MultiValueMode`, when we address multi-valued indexing/searching. So let's keep the code where we can find it until then.